### PR TITLE
Update code as per Julia's community guidelines

### PR DESCRIPTION
**Pull request by eulerkochy**
Multiline functions should use `return` for returning values.
PR is trivial enough.
I'll go through each of the files and try to spot if some part of the code violates standard. Gonna take a while. Bare through that period.

List of checked files.
- src/heaps
- robin_dict.jl

### CODE REVIEW

**Comment by oxinabox** [comment]
Looks great.
Is it ready to merge, or do you want to finish going through othter files?

**Comment by eulerkochy** [comment]
I'll plan to go through all the files and bump this PR. I have some commits
on local machine, I'll push them as soon as I can.

**Comment by eulerkochy** [comment]
@oxinabox I guess this PR is now ready! Have a look at it.

Now, since I read and understood through the entirity of the codebase(only the src, not the tests and benchmarks), here are few of my observations:
- code can be made more terse in some places. If it wasn't begging to be done, I haven't changed it in this PR, because I guess it was outside its scope.
- two files, `int_set.jl` and `trie.jl` are poorly documented in terms of docstrings.
- I absolutely loved the documentation of `sorted_dict.jl` and `sorted_multi_dict.jl`. What's so special about it? The docstrings are written in a nice way and most importantly, explicitly mention **asymptotic complexity of the operations**, something which is missing in the entire codebase. As a data structures repo, I feel this is quintessential and got to be present. Opinion?

**Comment by oxinabox** [comment]
> I absolutely loved the documentation of sorted_dict.jl and sorted_multi_dict.jl. What's so special about it?

Different parts of this package have come from different places at different times.
For example the priority queue and minheap stuff was written by Jeff back in Julia 0.1 days, and used to be part of the language itself.

The Sorted* types were written by @StephenVavasis

File: src/accumulator.jl [hunk 1]
```diff
@@ -33,7 +33,7 @@ function counter(seq)
 end
 
 eltype_for_accumulator(seq::T) where T = eltype(T)
-function eltype_for_accumulator(seq::T) where {T<:Base.Generator}
+function eltype_for_accumulator(seq::T) where {T <: Base.Generator}
```
Comment (oxinabox) [inline src/accumulator.jl]:
```suggestion
function eltype_for_accumulator(seq:: Base.Generator}
```

File: src/circular_buffer.jl [hunk 2]
```diff
@@ -38,11 +38,7 @@ end
 @inline function _buffer_index(cb::CircularBuffer, i::Int)
     n = cb.capacity
     idx = cb.first + i - 1
-    if idx > n
-        idx - n
-    else
-        idx
-    end
+    return ifelse(idx > n, idx - n, idx)
```
Comment (oxinabox) [inline src/circular_buffer.jl]: changing `if` to `ifelse` changes behavour.
Which could be a valid call in this case, but is out of scope for this PR which is about changing style
`?` does not change behavour
```suggestion
    return idx > n ? idx - n : idx
```

**Comment by oxinabox** [comment]
Just a few minor things.
Fix them and then it is good to merge.


---

I have invited you as a collaborator to the repository (via making you a member of the org).
you'll need to accept your invite and (if you haven't already) setup 2FA.
Few things:
 - Don't merge your own PRs without review. (Exception if just bumping version number to tag release.)
 - Follow the [Continuous Delivery pattern](https://white.ucc.asn.au/2019/09/28/Continuous-Delivery-For-Julia-Packages.html) and tag a release after every PR that has a functional change. Unless the project.toml version is suffixed with `-DEV`. Try and avoid making breaking releases. (thats what setting -DEV is for, bundling a collection of breaking changes)

File: src/circular_buffer.jl (hunk 2 above)
Comment (eulerkochy) [reply src/circular_buffer.jl]: true .. this was un-solicited change .. thanks for pointing it out

File: src/accumulator.jl [hunk 3]
```diff
@@ -33,7 +33,7 @@ function counter(seq)
 end
 
 eltype_for_accumulator(seq::T) where T = eltype(T)
-function eltype_for_accumulator(seq::T) where {T <: Base.Generator}
+function eltype_for_accumulator(seq:: Base.Generator}
```
Comment (oxinabox) [inline src/accumulator.jl]: my bad, added an extra space
```suggestion
function eltype_for_accumulator(seq::Base.Generator}
```

**Comment by eulerkochy** [comment]
Travis CI down ? Literally haven't changed anything for the tests to fail

**Pull request merged**